### PR TITLE
Generate a test plan from a sequence of events

### PR DESCRIPTION
**Pull request by Silverwolf90**
The goal of this PR is to use a test model by "manually" defining a sequence of events as opposed to generating them with `getSimplePathPlans`/`getShortestPathPlans`.

```js
dieHardModel.getPlanFromEvents(
  [
    { type: 'FILL_5' },
    { type: 'POUR_5_TO_3' },
    { type: 'EMPTY_3' },
    { type: 'POUR_5_TO_3' },
    { type: 'FILL_5' },
    { type: 'POUR_5_TO_3' }
  ],
  {
    target: 'success'
  }
)
```

This PR adds a `getPathFromEvents(machine, events): StatePath` function which is used by a new method `testModel.getPlanFromEvents(events, options): TestPlan`.

### CODE REVIEW

File: docs/packages/xstate-graph/index.md [hunk 1]
```diff
@@ -288,6 +288,81 @@ console.log(simplePaths);
...
+    closed: {
+      type: 'final'
+    }
+  }
+});
+
+const path = getPathFromEvents(feedbackMachine, [
+  { type: 'CLICK_GOOD' },
+  { type: 'SUBMIT' },
+  { type: 'CLOSE },
```
Comment (davidkpiano) [inline docs/packages/xstate-graph/index.md:343]:
```suggestion
  { type: 'CLOSE' },
```

File: packages/xstate-test/src/index.ts [hunk 2]
```diff
@@ -102,6 +103,28 @@ export class TestModel<TTestContext, TContext> {
...
+
+    if (!path.state.matches(target)) {
+      throw new Error(
+        `The last state ${JSON.stringify(
+          (path.state as any).value
+        )} does not match the target: ${JSON.stringify(target)}`
+      );
+    }
+
+    return this.getTestPlans({
```
Comment (Silverwolf90) [inline packages/xstate-test/src/index.ts:120]: Should I return an array here?

File: packages/xstate-graph/src/graph.ts [hunk 3]
```diff
@@ -381,3 +382,66 @@ export function toDirectedGraph(stateNode: StateNode): DirectedGraphNode {
...
+>(
+  machine: StateMachine<TContext, any, TEvent>,
+  events: Array<TEvent>
+): StatePath<TContext, TEvent> {
+  const optionsWithDefaults = getValueAdjMapOptions<TContext, TEvent>({
+    events: events.reduce((events, event) => {
+      events[event.type] ??= [];
+      events[event.type].push(event);
+      return events;
+    }, {})
```
Comment (Silverwolf90) [inline packages/xstate-graph/src/graph.ts:398]: This doesn't do anything about duplicates, figured it doesn't matter. But wanted to highlight.

File: packages/xstate-test/src/index.ts (hunk 2 above)
Comment (Silverwolf90) [reply packages/xstate-test/src/index.ts]: after sleeping on it, decided to go with the single plan

**Review by davidkpiano — APPROVED** [review]
Really nice PR; this is going to be very useful 👍

File: packages/xstate-test/src/index.ts (hunk 2 above)
Comment (davidkpiano) [reply packages/xstate-test/src/index.ts]: Yep, it's made obvious by the name

**Pull request merged**